src/libstd/collections/hash/map.rs

```diff
@@ -382,12 +400,15 @@ fn robin_hood<'a, K: 'a, V: 'a>(mut bucket: FullBucketMut<'a, K, V>,
     // There can be at most `size - dib` buckets to displace, because
     // in the worst case, there are `size` elements and we already are
     // `distance` buckets away from the initial one.
-    let idx_end = starting_index + size - bucket.distance();
+    let idx_end = starting_index + size - bucket.displacement();
 
     loop {
-        let (old_hash, old_key, old_val) = bucket.replace(hash, k, v);
+        let (old_hash, old_key, old_val) = {
```

Comment:
I tried to get rid of non-essential methods in the `table` module. Besides, `replace` was used only from here.